swim/join_sender.go

```diff
@@ -73,6 +77,10 @@ type joinSender struct {
 	node    *Node
 	timeout time.Duration
 
+	// bootstrapHostsMap is a map of unique hosts each containing a slice of
+	// the instances (hostsports) on that particular host.
+	bootstrapHostsMap map[string][]string
```

Comment:
Grouping by host should be an implementation detail of a Bootstrap Selector-type-of-abstraction.